src/testsuite.jl

```diff
@@ -49,6 +49,7 @@ end
 Runs the entire GPUArrays test suite on array type `AT`
 """
 function test(AT::Type{<:GPUArray})
+    GPUArrays.allowscalar(false)
```

Comment:
my reasoning here was that implementations can decide themselves if they want to assert scalar indexing, eg. https://github.com/JuliaGPU/CuArrays.jl/blob/master/test/runtests.jl#L27
but then it probably needs to be added to the JLArray tests.

Reply:
Then maybe moving scalar indexing assertion into runtest.jl of GPUArrays package for JLArray tests, like
```julia
@testset "JLArray" begin
    local _prev = GPUArrays._allowscalar[]
    GPUArrays.allowscalar(false)
    GPUArrays.test(JLArray)
    GPUArrays._allowscalar[] = _prev
end
```

Reply:
Something like that, yeah. But we can leave that for a different PR.

Reply:
(something like `@allowscalar`, eg. `@disallowscalar` would be cleaner then, not having to work with unexported values in GPUArrays)